nrf_802154_sl/CMakeLists.txt

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# Calculates relative path to the library to be uses taking into account
+# build type of the library, SoC type, floating point ABI, supported FEM type.
+# Note: Subdirectory layout is specific to nrf_802154_sl lib
+function(calculate_lib_path lib_path)
+  # Select build type of the library to link with
+  set(build_type_dir "Release")
```

Comment:
What about "Debug" libs I've seen earlier? If they're not included, perhaps there's no need for this subdirectory in public repository?

Reply:
Consider if we would like to provide Debug versions to one of our customers. It would be so easy to tell them "copy files to nrfxlib/nrf_802154_sl/lib/Debug" and change build_type_dir to "Debug". That's it. If structure of directories is more flat it will be more difficult.

Reply:
I'd like to hear from other reviewers, quite unprecedented approach, but I see your motivation now.

Reply:
For the MPSL and BLE controller libraries we currently do not want to expose the debug libraries. If we want to do that in the future, I'm not sure if it is better to deliver both a debug and release lib, or just providing a debug lib in a separate branch/fork.

Reply:
We could deliver 802.15.4 "Debug" libraries publicly, or on-demand with NDA or other procedure (it is not decided yet).
The point is that the customer, that would use "Debug" libraries in a project, should have an easy way to switch between Release and Debug versions when building. In my opinion, keeping `Release` and `Debug` directories side by side is the most popular and straightforward way of handling this problem. Switching between repositories or branches would be more error-prone for a customer, what could unnecessarily increase support cost for such customers.
Because of that I would suggest to keep the `Release` directory here to keep space to enable `Debug` builds of the customers projects (regardless distribution method of `Debug` version we choose).

Reply:
What does `Debug` build mean in this case.
In CMake, the term `CMAKE_BUILD_TYPE` exists, which supports `RELEASE` and `DEBUG`, but this is not used in Zephyr, as it is not compatible with Kconfig.
So in Zephyr we have no notion of Debug vs. Release.

But here we introduce a `Debug` concept foir switching libs, but there is no config option.

I would leave out this for this PR, and if this is a highly needed feature, then let's work on a common approach for a later release.

If you need to give a debig lib to a customer, he can just overwrite the existing lib.
Afterall, with the suggested solution in this PR, he would anyway need make changes to the CMakeLists.txt file.

Reply:
Structure flattened Instead of `Debug/${soc_dir}/${float_dir}/${fem_type}`, there is only
`${soc_dir}/${float_dir}`